// frontity/dist/src/commands/dev.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const chalk_1 = __importDefault(require("chalk"));
const utils_1 = require("../utils");
const choosePort_1 = __importDefault(require("../utils/choosePort"));
const HOST = process.env.HOST || "0.0.0.0";
exports.default = async ({ production, port, https, target, dontOpenBrowser, publicPath, }) => {
    let dev;
    const options = {
        mode: production ? "production" : "development",
        port: parseInt(port, 10) || 3000,
        isHttps: !!https,
        target: target || "module",
        openBrowser: !dontOpenBrowser,
        publicPath: publicPath || "/static/",
    };
    try {
        dev = require("@frontity/core").dev;
    }
    catch (error) {
        const message = `Make sure that you are running ${chalk_1.default.green("frontity")} inside a Frontity project.\n` +
            `If so try installing ${chalk_1.default.green("@frontity/core")} again with ${chalk_1.default.green("npm i @frontity/core")}.\n`;
        utils_1.errorLogger(error, message);
    }
    try {
        const port = await choosePort_1.default(HOST, options.port);
        if (port === null) {
            return;
        }
        await dev(Object.assign(Object.assign({}, options), { port }));
    }
    catch (error) {
        utils_1.errorLogger(error);
    }
};
